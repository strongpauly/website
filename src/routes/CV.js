import React, { Component } from "react";
import { HiFrame } from "../components/HiFrame";

export class CV extends Component {
  render() {
    return (
      <div id="cvContainer">
        <p>
          CV implemented using react, react-bootstrap and{" "}
          <a href="https://jsonresume.org/schema/">jsonresume</a>. It&apos;s
          source can be found{" "}
          <a href="https://www.github.com/strongpauly/cv">here</a>.
        </p>
        <HiFrame src="/cv/index.html" title="cv"></HiFrame>
      </div>
    );
  }
}
