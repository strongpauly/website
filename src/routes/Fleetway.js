import React, { Component } from 'react';
import Script from 'react-load-script';

export class Fleetway extends Component {

  render() {
    return <div id="fleetwayContainer" className="container">
      <p>This is a injectable search bar that I developed for a client Fleetway travel.  It downloads an injector js file which then
      requests other css and javascript files to build the angular component.  Executing the search will take you to their site.</p>
      <div style={{minHeight: '150px', marginTop: '20px', marginBottom: '20px'}}>
        <search-widget></search-widget>
        <Script
          url="https://bookings.fleetwaytravel.com/ng/injector.js?aid=fw"
          attributes={{id: 'travelsmartInjector'}} />
      </div>
      <p>The injector.js is vanilla javascript that creates additional script and css tags once it has been loaded.  Fleetway partner with other
        providers, each of which are given a particular identifier.  The injector.js also downloads an encrypted version of this for use in
        further api calls.
      </p>
      <p>The search bar itself is a combination of Angular 4 bootstrap and custom components, including a type ahead search, date range picker and
      other drop down controls.  It was designed to be flexible and reusable component.  It is included not only as an injectable widget, but also
      on the fleetway home page, as well as on multiple pages throughout their booking platform.</p>
    </div>;
  }
}
