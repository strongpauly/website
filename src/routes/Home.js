import React, { Component } from 'react';

export default class Home extends Component {

  render() {
    return <div id="homeContainer" className="container"><p>This is a portfolio of projects used in investigations into technologies by London based, full stack developer, Paul Potsides.</p>
      <p>Currently, this is mainly client side and react based projects, but I will hopefully be expanding this into some Java or Node based middle tier projects too.</p>
      <p>This site was built using react, react-bootstrap and react-router.  It&apos;s source can be found <a href="https://www.github.com/strongpauly/website">here</a>
      </p>
    </div>;
  }

}
