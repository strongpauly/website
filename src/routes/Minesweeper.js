import React, { Component } from 'react';
import HiFrame from '../components/HiFrame';

export class Minesweeper extends Component {

  render() {
    return <div id="minesweeperContainer">
      <div className="container">
        An implementation of the classic game using react. It&apos;s source can be found <a href="https://www.github.com/strongpauly/minesweeper">here</a>.
      </div>
      <HiFrame src="/minesweeper" title="minesweeper"></HiFrame>
    </div>;
  }

}
