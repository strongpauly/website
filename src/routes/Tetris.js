import React, { Component } from 'react';
import {HiFrame} from '../components/HiFrame';

export class Tetris extends Component {

  render() {
    return <div id="tetrisContainer">
      <p>
        The classic Gameboy game implemented with react and redux.
        It&apos;s source can be found <a href="https://github.com/strongpauly/tetris">here</a>.
      </p>
      <HiFrame src="/tetris" title="tetris"></HiFrame>
    </div>;
  }

}
