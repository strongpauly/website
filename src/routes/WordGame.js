import React, { Component } from 'react';
import {HiFrame} from '../components/HiFrame';

export class WordGame extends Component {

  render() {
    return <div id="wordGameContainer">
      <p>
        A simple word game implemented using react.  Select words of the length suggested by the box below the grid.
        <ul>
          <li>The + will start a new game.</li>
          <li>The ♺ will reset the game.</li>
          <li>The ? will provide a hint. </li>
        </ul>
        If the game can no longer be completed, the ♺ will rotate.
        It&apos;s source can be found <a href="https://www.github.com/strongpauly/wordgame">here</a>.
      </p>
      <HiFrame src="/wordgame" title="wordgame"></HiFrame>
    </div>;
  }

}
