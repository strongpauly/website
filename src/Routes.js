import React, { Component } from 'react';
import {Row} from 'react-bootstrap';
import {Route, Redirect} from 'react-router-dom';

//Routes
import Home from './routes/Home';
import CV from './routes/CV';
import Minesweeper from './routes/Minesweeper';
import WordGame from './routes/WordGame';
import Tetris from './routes/Tetris';

class Routes extends Component {

  render() {
    return (
      <Row>
        <Redirect from="/" exact to="/r/home" />
        <Route path="/r/home" component={Home} />
        <Route path="/r/cv" component={CV} />
        <Route path="/r/minesweeper" component={Minesweeper} />
        <Route path="/r/wordgame" component={WordGame} />
        <Route path="/r/tetris" component={Tetris}/>
      </Row>
    );
  }
}

export default Routes;
