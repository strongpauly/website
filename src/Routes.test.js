import React from 'react';
import { shallow } from 'enzyme';
import { Route } from 'react-router';
import Routes from './Routes.js';
import Home from './routes/Home';
import CV from './routes/CV';
import Minesweeper from './routes/Minesweeper';
import WordGame from './routes/WordGame';
import Tetris from './routes/Tetris';
import Fleetway from './routes/Fleetway';
/* eslint-env jest */

describe('Routes', () => {
  it('renders correct routes', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
    const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.component;
      return pathMap;
    }, {});
    // { 'r/home' : Home, ... }
    expect(pathMap['/r/home']).toBe(Home);
    expect(pathMap['/r/cv']).toBe(CV);
    expect(pathMap['/r/minesweeper']).toBe(Minesweeper);
    expect(pathMap['/r/wordgame']).toBe(WordGame);
    expect(pathMap['/r/tetris']).toBe(Tetris);
    expect(pathMap['/r/fleetway']).toBe(Fleetway);
  });
});
