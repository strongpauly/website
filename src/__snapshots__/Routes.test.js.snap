// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders correct routes 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={12}
  >
    <Switch>
      <Redirect
        exact={true}
        from="/"
        push={false}
        to="/r/home"
      />
      <Route
        component={[Function]}
        path="/r/home"
      />
    </Switch>
    <Route
      component={[Function]}
      path="/r/cv"
    />
    <Route
      component={[Function]}
      path="/r/oss"
    />
    <Route
      component={[Function]}
      path="/r/minesweeper"
    />
    <Route
      component={[Function]}
      path="/r/wordgame"
    />
    <Route
      component={[Function]}
      path="/r/tetris"
    />
  </Col>
</Row>
`;
